Setting LD_LIBRARY_PATH is a really bad idea when cross compiling. Simplest way
to disable this is to rename the variable as per the patch below, then its 
harmless. This is a hack.

RP 10/6/2010

Upstream-Status: Inappropriate [Other]
Hacky workaround not likely to be accepted upstream.

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

Index: source/config/Makefile.inc.in
===================================================================
--- source.orig/config/Makefile.inc.in	2010-06-10 21:49:19.000000000 +0100
+++ source/config/Makefile.inc.in	2010-06-10 21:49:46.000000000 +0100
@@ -126,7 +126,7 @@
 ##################################################################
 # Environment variable to set a runtime search path
 #   (Overridden when necessary in -mh files)
-LDLIBRARYPATH_ENVVAR = LD_LIBRARY_PATH
+LDLIBRARYPATH_ENVVAR = LD_LIBRARY_PATH2
 
 # Versioned target for a shared library
 FINAL_SO_TARGET = $(SO_TARGET).$(SO_TARGET_VERSION)
Index: source/icudefs.mk.in
===================================================================
--- source.orig/icudefs.mk.in	2010-06-10 21:51:03.000000000 +0100
+++ source/icudefs.mk.in	2010-06-10 21:51:09.000000000 +0100
@@ -150,7 +150,7 @@
 SHLIB.cc=     $(CXX) $(CXXFLAGS) $(LDFLAGS) -shared
 
 # Environment variable to set a runtime search path
-LDLIBRARYPATH_ENVVAR = LD_LIBRARY_PATH
+LDLIBRARYPATH_ENVVAR = LD_LIBRARY_PATH2
 
 # Versioned target for a shared library.
 FINAL_SO_TARGET = $(SO_TARGET).$(SO_TARGET_VERSION)
Index: source/tools/genrb/gendtjar.pl
===================================================================
--- source.orig/tools/genrb/gendtjar.pl	2010-06-10 21:51:36.000000000 +0100
+++ source/tools/genrb/gendtjar.pl	2010-06-10 21:52:03.000000000 +0100
@@ -68,8 +68,8 @@
         $icuLibDir = abs_path($icuBinDir."/../lib");
         $path .=":$icuBinDir:$icuLibDir";
         
-        $libpath = $ENV{'LD_LIBRARY_PATH'}.":$icuLibDir";
-        $ENV{'LD_LIBRARY_PATH'} = $libpath;
+        $libpath = $ENV{'LD_LIBRARY_PATH2'}.":$icuLibDir";
+        $ENV{'LD_LIBRARY_PATH2'} = $libpath;
         
         #print ("#####  LD_LIBRARY_PATH = $ENV{'LD_LIBRARY_PATH'}\n");
     
